src/Policy/OrmResolver.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization\Policy;
+
+use Authorization\Policy\Exception\MissingPolicyException;
+use Cake\Core\App;
+use Cake\Datasource\EntityInterface;
+use Cake\Datasource\RepositoryInterface;
+use InvalidArgumentException;
+
+/**
+ * Policy resolver that applies conventions based policy classes
+ * for CakePHP ORM Tables and Entities.
+ */
+class OrmResolver implements ResolverInterface
+{
+    /**
+     * Application namespace.
+     *
+     * @var string
+     */
+    protected $appNamespace;
+
+    /**
+     * Constructor
+     *
+     * @param string $appNamespace The application namespace
+     */
+    public function __construct($appNamespace = 'App')
+    {
+        $this->appNamespace = $appNamespace;
+    }
+
+    /**
+     * Get a policy for an ORM Table or Entity.
+     *
+     * @param \Cake\ORM\RepositoryInterface|\Cake\Datasource\EntityInterface $resource The resource.
+     * @return object
+     * @throws \InvalidArgumentException When a resource is not an ORM object.
+     * @throws \Authorization\Policy\Exception\MissingPolicyException When a policy for the
+     *   resource has not been defined.
+     */
+    public function getPolicy($resource)
+    {
+        if ($resource instanceof EntityInterface) {
+            return $this->getEntityPolicy($resource);
+        }
+        if ($resource instanceof RepositoryInterface) {
```

Comment:
Should we support `QueryInterface` as well? This would be helpful for decorating a query object.  The policy could be resolved based on the query repository.

Reply:
I was considering it. Would you expect the policy to resolve to the table policy, or something else?

Reply:
Maybe this could resolve to `ArticlesQuery` policy and fall back to `ArticlesTable`. Would that make sense or bring too much confusion?

Reply:
We could have `ArticlesQueryPolicy`. Relying on policies to mutate queries in place feels kind of gross though. In pundit 'query policies' are a special type of policy that accept a query and return a new query with conditions applied. An example would be.

```ruby
query = policy_scope(Bookmarks.find).resolve
```

The `resolve` method update the query with additional conditions based on the current user.

Reply:
Maybe we should have another method in the service for resource modification based on policy rules? Running a query through `can()` method in order to inject some conditions doesn’t seem intuitive.

Reply:
I agree that `can()` mutating state is icky feeling. I can draft up some ideas and post them in an issue later this week.